### ISSUE TITLE
We need guides.

### ISSUE THREAD

**Issue by erquhart** [netlify/netlify-cms#2092]
We need to write a whole lot of guides and pile them into the docs.

Let's keep a list of topics here. Suggest topics via comments, use reactions to +1 your favorites.

- [ ] - Controlling Filenames

**Comment by tech4him1** [comment]
I think we should also create a list of "common" errors and how to solve. For example, Git Gateway not being turned on. Would a separate issue be better for this?

**Comment by talves** [comment]
I have even envisioned the errors send a url that links to the docs to explain how to fix or read about that part of the configuration.

**Comment by erquhart** [comment]
@tech4him1 that would be a solid guide, maybe call it Troubleshooting? Or something? But yes yes yes.

@talves very much agreed, but should go to a separate issue. The goal if this one is to end up with just a ton of static guides, so that the only work required is writing documentation.

**Comment by krismorf** [comment]
Claimed the Gatsby guide

**Comment by adamwatters** [comment]
Just started working on one for Jekyll

**Comment by erquhart** [comment]
Awesome!!

**Comment by steffenz** [comment]
Hi! I just built a site with NextJS and Netlify CMS, and would love to give something back (and get to know the community) by contributing with a guide! 😁Is it still "up for grabs"?

**Comment by talves** [comment]
@steffenz yes. You can definitely start working on it. Just start it and do a PR with [WIP][Guide] in the beginning of the title and we will mark it as being worked on.

Thanks for contributing!

**Comment by tomrutgers** [comment]
Started working on a middleman guide

**Comment by adamwatters** [comment]
Finally got to the Jekyll guide. I'm planning on giving it a second pass in the coming days, but should be ready for another set of eyes if anyone wants to help with revision.

**Comment by morkro** [comment]
Why only documenting site generators? I am successfully using Netlify-CMS with a plain Vue.js and `vue-i18n` integration*. Would it be of interest to also provide guides for these implementations?

\* https://github.com/morkro/euromat

**Comment by erquhart** [comment]
@morkro a plain Vue guide would be fantastic!

**Comment by nikosolihin** [comment]
Just taught a workshop with Eleventy and Netlify CMS. Would love to claim the guide for it!

**Comment by erquhart** [comment]
Have at it!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jballe** [comment]
I could probably write a guide on using Hugo

**Comment by erquhart** [comment]
That would be great!

**Comment by cdvillard** [comment]
Gridsome has a plugin with documentation on how to install Netlify CMS. Would that suffice or would a local doc be more appropriate?

**Comment by tomrutgers** [comment]
@cdvillard We'd like to add it to our own docs so people can find their way to it within the netlify project. Converting the [Gridsome documentation](https://gridsome.org/docs/guide-netlify-cms/) to a docs entry within the project shouldn't take much effort I suppose

**Comment by carvajalluis** [comment]
Hello, i'm unable to follow the guide to use nextjs

![image](https://user-images.githubusercontent.com/674024/62243511-06678700-b3a3-11e9-9078-01e9d78d3ac9.png)
 cant find a way to use the loader 

  can you be more specific about versions and maybe have it in the exaple folder ?
this is my packa ge json, anithing else i tripple checked it is the same .
```json
{
  "scripts": {
    "dev": "next",
    "build": "next build && next export",
    "start": "serve ./out"
  },
  "dependencies": {
    "next": "^9.0.2",
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "frontmatter-markdown-loader": "^1.8.0"
  }
}
```

**Comment by tomrutgers** [comment]
@steffenz could you comment on that?

**Comment by xdesro** [comment]
Hey folks! Just did a basic guide for Hugo as well! Feedback welcome, as always. Hope it's helpful!

**Comment by erquhart** [comment]
Thank you Henry!!

**Comment by adamayd** [comment]
I'll claim the VuePress doc if no one else has it.  Also @cdvillard and @tomrutgers was the Gridsome doc spoken for or was that just a conversation about how to do it?  I could claim that one as well if it's needed.

**Comment by erquhart** [comment]
Both are up for grabs, that would be awesome!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by bingr001** [comment]
@erquhart  Can I work on the Using with CSS-in-JS ?

**Comment by ZoltanVeres** [comment]
I'm surprised we don't have an Eleventy Guide yet, i'm planning on working on it, because i need it for my personal blog. 
@nikosolihin  **Are you working on this or is it up for grabs?**
I identified two eleventy starters where Netlify CMS is in use, here are the changes related to NetlifyCMS:
https://github.com/danurbanowicz/eleventy-netlify-boilerplate/commits?author=biilmann
https://github.com/hankchizljaw/hylia/commits?author=erquhart

The setup used on Hylia is much sophisticated (made by @erquhart ) and it's closer to my needs but the builerplate starter (by @biilmann )seems easier to digest which is closer to what we have for the other starters. So i'm not sure which route should i go. 
Any thoughts @erquhart ?

**Comment by erezrokah** [comment]
How about adding a guide on how to run the CMS without Netlify Hosting:
https://github.com/netlify/netlify-cms/issues/2695
https://github.com/netlify/netlify-cms/issues/2774

**Comment by umutykaya** [comment]
@erezrokah It could be great to have a guide for working with aws amplify.

**Comment by mdartic** [comment]
I don't see any guide started for VuePress.
I could make a PR to create this guide.
@erquhart  do you have a guide to create a guide ?
I can create one from ideas of others SSG, if it's ok for you.

**Comment by adamayd** [comment]
@mdartic I failed miserably on my VuePress attempt then lost track of it, please take over.

**Comment by ourlaw** [comment]
docusaurus

**Comment by dsanderson90** [comment]
Can we have a section dedicated to registerEditorComponent? I am looking for docs that explain the fromBlock function and can't find anything beyond this https://www.netlifycms.org/docs/custom-widgets/#registereditorcomponent

**Comment by DhimanSarkar** [comment]
### Any one has any detailed documentation on **_registerEditorComponent_**??

**Comment by Alexa-Green** [comment]
> docusaurus

@massoudmaboudi Are you working on the docusaurus guide or should I?

**Comment by erquhart** [comment]
@Alexa-Green I think that was just a suggestion, I'd say go for it

**Comment by silverwebb925** [comment]
@erquhart I'm very eager to contribute to the Docs, both in terms of add/updating content and improving the CMS configuration so it is easier to work with via Open Authoring.
Put in my first PR (ever!) with an improvement to the section organization workflow, and would appreciate some help to get up to speed with the contribution process if you (or someone else) has time to comment. Also posting to the Slack channel shortly.

[PR #4250](http://google.com)https://github.com/netlify/netlify-cms/pull/4250)

**Comment by emileswain** [comment]
I second @d-sanderson suggestion regards a guide for custom components that cover a lot more detail that the current guide provides. 

Attempting to go beyond 'hello world' instantly chucks developers into a rabbit hole of despair.
Issues i've yet to resolve (not quite ready to just give up yet) 
1. Properly deploy custom widget to github repo and include in cms (before deploying to npm).
2. Create custom editor widget that embeds react code into markdown. (various issues and tests of patience yet to be resolved)
3. Not immediately obvious how to register custom components (I'm on NextJS static site gen).
4. Create custom widget**s** that work together. I can get a single markdown widget to work, but now it conflicts with the others. So there is clearly some magic with the regex detecting the widget code that i'm doing wrong, but there is zero documentation on this. 

When you see a component someones written you like and bring it into your netlify CMS package, it becomes incumberant on your build process. this makes it very hard to leverage your components as a guide for custom components of our own. 

Its very disparaging.

**Comment by beriberikix** [comment]
@Alexa-Green or @erquhart were either of you still planning on writing a docusaurus guide?

**Comment by aentwist** [comment]
As JavaScript framework churn continues, and as a result of the highly divisive Vue 3 release, I suggest dropping [VuePress](https://vuepress.vuejs.org/) in favor of [VitePress](https://vitepress.dev). FYI

> However, maintaining two SSGs in parallel isn't sustainable, so the Vue team has decided to focus on VitePress as the main recommended SSG in the long run.

See [What About VuePress?](https://vitepress.dev/guide/what-is-vitepress#what-about-vuepress)
